feat(billing): Disable submission form button when adding/removing credits

dashed opened the pull request:
Closes https://linear.app/getsentry/issue/BIL-232/credit-to-a-subscription-can-sometimes-get-duplicated-via-admin

 Disable submission form button when adding/removing credits. This should prevent accidental double-clicks to add multiple credits.

dashed committed:
feat(billing): Disable submission form when adding/removing credits

dashed committed:
fix test

ajay-sentry reviewed (approved):
nothing blocking, just a couple thoughts

dashed committed:
ref(billing): Use submitDisabled prop to disable submit button

Add submitDisabled prop to Form component to properly disable the
submit button during form submission, utilizing the built-in Form API.
Also update tests to verify submit button disabled state.
